### PR TITLE
Add AGENTS.md

### PR DESCRIPTION
**Pull request by yiyixuxu**

### CODE REVIEW

File: .ai/AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Diffusers — Agent Guide
+
+### Philosophy
+
+Write code as simple and explicit as possible.
```
Comment (sayakpaul) [inline .ai/AGENTS.md]:
```suggestion
Strive to write code as simple and explicit as possible.
```

File: .ai/AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        self.to_q = nn.Linear(query_dim, heads * dim_head, bias=False)
+        self.to_k = nn.Linear(query_dim, heads * dim_head, bias=False)
+        self.to_v = nn.Linear(query_dim, heads * dim_head, bias=False)
+        self.to_out = nn.ModuleList([nn.Linear(heads * dim_head, query_dim), nn.Dropout(0.0)])
+        self.set_processor(MyModelAttnProcessor())
+
+    def forward(self, hidden_states, attention_mask=None, **kwargs):
+        return self.processor(self, hidden_states, attention_mask, **kwargs)
+```
+
```
Comment (sayakpaul) [inline .ai/AGENTS.md:67]:
```suggestion

Consult the implementations in `src/diffusers/models/transformers/` if you need further references.

```

File: Makefile [hunk 3]
```diff
@@ -98,3 +98,14 @@ post-release:
...
 post-patch:
 	python utils/release.py --post_release --patch
+
+# AI agent symlinks
+
+codex:
+	ln -snf .ai/AGENTS.md AGENTS.md
+
+claude:
+	ln -snf .ai/AGENTS.md CLAUDE.md
```
Comment (sayakpaul) [inline Makefile:108]: Unclear about what the purpose of symlinking here 👀

File: .ai/AGENTS.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+### Pipeline
+- All pipelines must inherit from `DiffusionPipeline`
+
+### Tests
+- Slow tests gated with `@slow` and `RUN_SLOW=1`
```
Comment (sayakpaul) [inline .ai/AGENTS.md]:
```suggestion
- Slow tests gated with `@slow` and `RUN_SLOW=1`
- All model-level tests must use the `BaseModelTesterConfig`, `ModelTesterMixin`, `MemoryTesterMixin`, `AttentionTesterMixin`, `LoraTesterMixin`, and `TrainingTesterMixin` classes initially to write the tests. Any additional tests should be added after discussions with the maintainers. Use `tests/models/transformers/test_models_transformer_flux.py` as a reference.

```

File: .ai/AGENTS.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        self.to_v = nn.Linear(query_dim, heads * dim_head, bias=False)
+        self.to_out = nn.ModuleList([nn.Linear(heads * dim_head, query_dim), nn.Dropout(0.0)])
+        self.set_processor(MyModelAttnProcessor())
+
+    def forward(self, hidden_states, attention_mask=None, **kwargs):
+        return self.processor(self, hidden_states, attention_mask, **kwargs)
+```
+
+### Pipeline
+- All pipelines must inherit from `DiffusionPipeline`
```
Comment (sayakpaul) [inline .ai/AGENTS.md]:
```suggestion
- All pipelines must inherit from `DiffusionPipeline`. Consult implementations in `src/diffusers/pipelines` in case you need references.
- DO NOT use an existing pipeline class (e.g., `FluxPipeline`) to override another pipeline (e.g., `FluxImg2ImgPipeline` which will be a part of the core codebase (`src`).

```

File: .ai/AGENTS.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+- `make style` and `make fix-copies` should be run as the final step before opening a PR
+
+### Copied Code
+- Many classes are kept in sync with a source via a `# Copied from ...` header comment
+- Do not edit a `# Copied from` block directly — run `make fix-copies` to propagate changes from the source
+- Remove the header to intentionally break the link
+
+### Models
+- All layer calls should be visible directly in `forward` — avoid helper functions that hide `nn.Module` calls.
+- Attention must follow the diffusers pattern: both the `Attention` class and its processor are defined in the model file. The processor's `__call__` handles the actual compute and must use `dispatch_attention_fn` rather than calling `F.scaled_dot_product_attention` directly. The attention class inherits `AttentionModuleMixin` and declares `_default_processor_cls` and `_available_processors`.
```
Comment (sayakpaul) [inline .ai/AGENTS.md:28]:
```suggestion
- All layer calls should be visible directly in `forward` — avoid helper functions that hide `nn.Module` calls.
- Try to not introduce graph breaks as much as possible for better compatibility with `torch.compile`. For example, DO NOT arbitrarily insert operations from NumPy in the forward implementations.
- Attention must follow the diffusers pattern: both the `Attention` class and its processor are defined in the model file. The processor's `__call__` handles the actual compute and must use `dispatch_attention_fn` rather than calling `F.scaled_dot_product_attention` directly. The attention class inherits `AttentionModuleMixin` and declares `_default_processor_cls` and `_available_processors`.
```

File: .ai/AGENTS.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
+# Diffusers — Agent Guide
+
+### Philosophy
```
Comment (stevhliu) [inline .ai/AGENTS.md]: the rest of these headers should be second-level

```suggestion
## Coding style
```

File: .ai/AGENTS.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+- No defensive code or unused code paths — do not add fallback paths, safety checks, or configuration options "just in case". When porting from a research repo, delete training-time code paths, experimental flags, and ablation branches entirely — only keep the inference path you are actually integrating.
+- Do not guess user intent and silently correct behavior. Make the expected inputs clear in the docstring, and raise a concise error for unsupported cases rather than adding complex fallback logic.
+
+---
+
+### Dependencies
+- No new mandatory dependency without discussion (e.g. `einops`)
+- Optional deps guarded with `is_X_available()` and a dummy in `utils/dummy_*.py`
+
+### Code Style
```
Comment (stevhliu) [inline .ai/AGENTS.md]:
```suggestion
## Code formatting
```

**Comment by yiyixuxu** [comment]
@stevhliu 
yes, totally make sense. I have a `model-integration.md` I'm working on, will add in a follow-up PR and follow this pattern you suggested here.
Additionally, we will also have a skills folder to host some official skills, follow https://github.com/huggingface/transformers/pull/44489/changes

File: Makefile (hunk 3 above)
Comment (yiyixuxu) [reply Makefile:108]: so that it turn int CLAUDE.md for claude code users
